### ISSUE TITLE
Unable to get same transformation for image and mask.

### ISSUE THREAD

**Issue by athollywood** [albumentations-team/albumentations#715]
## 🐛 Bug

The same transformation is not applied to the image and mask when applied inside custom generator using keras.utils.Sequence.

## To Reproduce

1. Create data generator class that inherits the properties of the keras.utils.Sequence class.
2. Within generator class, perform image augmentation on both the image and mask using albumentation.Compose.

```python
class DataGen(keras.utils.Sequence):
    """Helper to iterate over the data (as Numpy arrays)."""

    def __init__(self, batch_size, img_size, input_img_paths, target_img_paths,augment):
        self.batch_size = batch_size
        self.img_size = img_size
        self.input_img_paths = input_img_paths
        self.target_img_paths = target_img_paths
        self.augment = augment
        

    def __len__(self):
        return len(self.target_img_paths) // self.batch_size

    def __getitem__(self, idx):
        """Returns tuple (input, target) correspond to batch #idx."""
        i = idx * self.batch_size
        batch_input_img_paths = self.input_img_paths[i : i + self.batch_size]
        batch_target_img_paths = self.target_img_paths[i : i + self.batch_size]
        x = np.zeros((batch_size,) + self.img_size + (3,), dtype="float32")
        for j, path in enumerate(batch_input_img_paths):
            img = np.array(load_img(path, target_size=self.img_size))/255
            x[j] = img
        y = np.zeros((batch_size,) + self.img_size, dtype="uint8")
        for j, path in enumerate(batch_target_img_paths):
            img = np.array(load_img(path, target_size=self.img_size, color_mode="grayscale"))
            y[j] = img
        
        if self.augment:
            for i in range(batch_size):
                x[i, ], y[i, ] = self.__random_transform(
                    x[i, ], y[i, ]) 
        
        y[y>2] = 2

        return x, y
        
    def __random_transform(self, img, mask):
        composition = albu.Compose([
            albu.VerticalFlip()
        ])
        composed = composition(image=img, mask=mask)
        aug_img = composed['image']
        aug_mask = composed['mask']

        return aug_img, aug_mask
```
3.  Insatiate data generator and display a few examples.
```python
random.Random(1337).shuffle(input_img_paths)
random.Random(1337).shuffle(target_img_paths)
train_input_img_paths = input_img_paths[:-val_samples]
train_target_img_paths = target_img_paths[:-val_samples]

# Instantiate data Sequences for each split
train_gen = DataGen(
    batch_size, img_size, train_input_img_paths, train_target_img_paths, True
)

for i in range(10):
    sample_image = train_gen[0][0][i] #first batch, ith image.
    sample_mask = train_gen[0][1][i] #first batch, ith mask.
    visualize(sample_image, sample_mask)
```
Some image/mask pairs match:
![image](https://user-images.githubusercontent.com/4474293/94968819-36d71380-04c7-11eb-9095-2253d401ccd5.png)

Many do not:
![image](https://user-images.githubusercontent.com/4474293/94968927-6a19a280-04c7-11eb-9bae-be2490c77e30.png)

## Expected behavior

The same transformation is applied to the image and mask.

## Environment

 - Albumentations version (e.g., 0.4.6):
 - Python version (e.g., 3.7.9):
 - OS (e.g., Windows):
 - How you installed albumentations (`conda`):
 - Any other relevant information:

## Additional context

I hope this just a typo or something trivial on my end, but I have been trying to fix this for hours. Any help appreciated!

**Comment by athollywood** [comment]
Similar to #713, if I pull the augmentation function out and perform augmentation directly on it, I get the expected results.  It seems we both have issues when using it within the data generator (or batch loader as he calls it).

**Comment by athollywood** [comment]
Turns out it was trivial. I was calling the generator twice to get the image and mask.
